packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -816,6 +816,89 @@ class AndroidCameraCameraX extends CameraPlatform {
     await cameraControl.setZoomRatio(zoom);
   }
 
+  @override
+  Future<Iterable<VideoStabilizationMode>> getVideoStabilizationSupportedModes(
+      int cameraId) async {
+    final List<int> modeIndexes =
+        await _getAvailableVideoStabilizationModeIndexes();
+
+    final List<VideoStabilizationMode> list = <VideoStabilizationMode>[];
```

Comment:
Can the mode indexes --> `VideoStabilizationMode ` mode translation happen in the `Camera2CameraInfo` class versus here? `Camera2CameraInfo. getAvailableVideoStabilizationModes` could return the mode directly and the plugin implementation would be a bit more simple.

Reply:
Indeed, I made that improvement and it looks tidier. 

The downside to this was that I got a map from control modes (the ones specific to Android) to Video Stabilization Modes (the ones common to this plugin) in Camera2CameraInfo and the reverse map, performed by _getControlVideoStabilizationMode(...) in AndroidCameraCameraX, which though simpler is less elegant for maintenance. Usually I like to keep mapping stuff together, as close as possible. 

So, to fix that, I also made AndroidCameraCameraX._getControlVideoStabilizationMode(...) static (which I could have done previously 😳) and moved it to Camera2CameraInfo.getControlVideoStabilizationMode(...). Let me know if you disagree and I'll revert it (though I do think it is easier to have related mapping code as close together as possible).

Reply:
We typically like to keep the Dart wrapper classes of the native CameraX classes as similar as possible, so `getControlVideoStabilizationMode` does kinda break that pattern. I apologize because I led you in the wrong direction here. What do you think of reverting it and keeping all of the logic in `AndroidCameraCamerax` to try to maintain that pattern? I think this solution is also ok but I do agree about your concerns on the maintenance.

Just one thing for either solution you choose--can we put the `CameraMetadata` codes (like `CONTROL_VIDEO_STABILIZATION_MODE_OFF `) in `camera_metadata.dart` so we can reference the codes vs the integers?

Reply:
Thanks for the request for the constants in CameraMetadata. I hadn't noticed that class and it was indeed dying to have those constants added to it. :-)

As for the dilemma of where to put the mapping code, I partially reverted those changes and restored all the mapping code back to AndroidCameraCameraX, but I still kept the simplification that your suggestion led me to introduce, so I think we ended with better code after these iterations.